initiator.go

```diff
@@ -27,67 +29,104 @@ func RunDKG(init *Init) ([]*Result, error) {
 	/*
-		DKG ceremony ...
+		reshare ceremonies ...
 	*/
-	_, _, _, err := ValidateResults(
-		signedReshare.Reshare.NewOperators,
-		withdrawalCredentials,
-		validatorPK,
-		fork,
-		signedReshare.Reshare.Owner,
-		signedReshare.Reshare.Nonce,
-		id,
-		results)
-	return results, err
+	for i, reshareMsg := range signedReshare.Messages {
+		_, _, _, err := ValidateResults(
+			reshareMsg.Reshare.NewOperators,
+			reshareMsg.Reshare.WithdrawalCredentials,
+			reshareMsg.Reshare.ValidatorPubKey,
+			reshareMsg.Reshare.Fork,
+			reshareMsg.Reshare.Owner,
+			reshareMsg.Reshare.Nonce,
+			id,
+			results[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+	return results, nil
 }
 
-func RunResign(
-	validatorPK []byte,
-	withdrawalCredentials []byte,
-	fork [4]byte,
-	signedResign *SignedResign,
-	proofs map[*Operator]SignedProof,
-	client eip1271.ETHClient,
-) ([]*Result, error) {
-	operators := maps.Keys(proofs)
-
-	id := NewID()
+func RunResign(signedResign *SignedResign) ([][]*Result, error) {
+	id, err := GetReqIDfromMsg(signedResign)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reqID: %w", err)
+	}
 
-	var results []*Result
+	var results [][]*Result
 	/*
 		DKG ceremony ...
 	*/
 
-	_, _, _, err := ValidateResults(
-		operators,
-		withdrawalCredentials,
-		validatorPK,
-		fork,
-		signedResign.Resign.Owner,
-		signedResign.Resign.Nonce,
-		id,
-		results)
-	return results, err
+	for i, resignMsg := range signedResign.Messages {
+		_, _, _, err := ValidateResults(
+			resignMsg.Operators,
+			resignMsg.Resign.WithdrawalCredentials,
+			resignMsg.Resign.ValidatorPubKey,
+			resignMsg.Resign.Fork,
+			resignMsg.Resign.Owner,
+			resignMsg.Resign.Nonce,
+			id,
+			results[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+	return results, nil
+}
+
+func GetMessageHash(msg interface{}) ([32]byte, error) {
+	hash := [32]byte{}
+	switch msg := msg.(type) {
+	case SSZMarshaller:
+		// Single message case
+		msgBytes, err := msg.MarshalSSZ()
+		if err != nil {
+			return hash, err
+		}
+		copy(hash[:], eth_crypto.Keccak256(msgBytes))
+	case []*ResignMessage:
+		msgBytes := []byte{}
+		for _, resign := range msg {
+			resignBytes, err := resign.MarshalSSZ()
+			if err != nil {
+				return hash, err
+			}
+			msgBytes = append(msgBytes, resignBytes...)
+		}
+		copy(hash[:], eth_crypto.Keccak256(msgBytes))
+	case []*ReshareMessage:
+		msgBytes := []byte{}
+		for _, reshare := range msg {
+			reshareBytes, err := reshare.MarshalSSZ()
+			if err != nil {
+				return hash, err
+			}
+			msgBytes = append(msgBytes, reshareBytes...)
+		}
+		copy(hash[:], eth_crypto.Keccak256(msgBytes))
+	default:
+		return hash, fmt.Errorf("unexpected message type: %T", msg)
+	}
+	return hash, nil
 }
 
-// NewID generates a random ID from 2 random concat UUIDs
-func NewID() [24]byte {
-	var id [24]byte
-	b := uuid.New()
-	copy(id[:12], b[:])
-	b = uuid.New()
-	copy(id[12:], b[:])
-	return id
+func GetReqIDfromMsg(instance interface{}) ([24]byte, error) {
```

Comment:
@MatusKysel 
I think we should also take this function from spec now.. This can be done while we do audit fixes